x86-16bit-second-stage.json

```diff
@@ -0,0 +1,20 @@
+{
+	"arch": "x86",
+	"cpu": "i386",
+	"data-layout": "e-m:e-p:32:32-p270:32:32-p271:32:32-p272:64:64-f64:32:64-f80:32-n8:16:32-S128",
+	"dynamic-linking": false,
+	"executables": true,
+	"linker-flavor": "ld.lld",
+	"linker": "rust-lld",
+	"llvm-target": "i386-unknown-none-code16",
+	"max-atomic-width": 64,
+	"position-independent-executables": false,
+	"disable-redzone": true,
+	"target-c-int-width": "32",
+	"target-pointer-width": "32",
+	"target-endian": "little",
+	"panic-strategy": "abort",
+	"os": "none",
+	"vendor": "unknown",
+	"relocation-model": "pie"
```

Comment:
> My understanding was that the resulting binaries for a such target spec will still have relocations that need to be applied before it can be executed.

AFAIK PIE will only include relocations when depending on dynamic libraries.

> When specifying a custom target, we could just switch the relocation model to static, however we obviously can't do that if we want to use x86_64-unknown-none target.

Using the static relocation model would prevent KASLR as it requires a fixed load location at link time.

Reply:
> AFAIK PIE will only include relocations when depending on dynamic libraries.

No, PIEs will often still use relocations even when not depending on dynamic libraries, presumably for things where rip relative addressing is not an option e.g. vtables.

> Using the static relocation model would prevent KASLR as it requires a fixed load location at link time.

I was talking about the bootloader, not the kernel.

Reply:
> No, PIEs will often still use relocations even when not depending on dynamic libraries, presumably for things where rip relative addressing is not an option e.g. vtables.

Right, didn't think of that.

> I was talking about the bootloader, not the kernel.

I see. For bios it is fine to use a static relocation model then.

Reply:
> I see. For bios it is fine to use a static relocation model then.

`x86_64-unknown-none` would still be nice to have because it's now a Tier 2 target, so it doesn't need `-Zbuild-std` and is therefore compatible with bindeps.